USB: General fixes split from PR 12682

finikorg opened the pull request:
General fixes, can be applied before PR #12682.

finikorg committed:
usb: Remove unused variable

Cleanup code removing unused variable.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
samples: cdc_acm: Remove unneeded config options

Cleanup prj.conf

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
samples: fxos8700-hid: Use HID_MOUSE_REPORT_DESC macro

Use API macro for mouse description.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

jfischer-phytec-iot commented on samples/subsys/usb/cdc_acm/prj.conf:
```diff
@@ -10,11 +10,3 @@ CONFIG_USB_DEVICE_LOG_LEVEL_ERR=y
 CONFIG_UART_INTERRUPT_DRIVEN=y
 CONFIG_UART_LINE_CTRL=y
-
-CONFIG_USB_COMPOSITE_DEVICE=n
```
That was useful for composite testing, I would be happy if it stays or please add overlay for composite configuration.

finikorg replied:
Do you mean you change `=n` to `=y` when testing? Maybe this can be kept better in private repository. I added test section with composite enabled in the last patch.

finikorg replied:
I can create overlay, this is better way.

finikorg committed:
samples: cdc_acm: Add composite test configuration

Add section for build sanity check configuration with
CONFIG_USB_COMPOSITE_DEVICE enabled.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
sample: cdc_acm: Add overlay configuration for composite device

Add configuration for creating composite USB device with CDC ACM and
Mass Storage functions.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

jfischer-no reviewed (approved):
thanks
